scripts/new_upstream_snapshot.py

```diff
@@ -0,0 +1,627 @@
+                f"{env.devel_distro}."
+            )
+            if (
+                input(
+                    f"Is this the first devel upload for {env.devel_distro} "
+                    "(y/N)? "
+                ).lower()
+                == "y"
+            ):
+                is_devel = True
+                is_first_devel_upload = True
+            elif (
+                input(
+                    f"Is this the first SRU for series {env.changelog_distro} "
+                    "(y/N)? "
+                ).lower()
+                == "y"
+            ):
+                is_devel = False
+                is_first_sru = True
+    return is_devel, is_first_devel_upload, is_first_sru
+
+
+def get_sru_bug(bug, no_sru_bug):
+    """Ask the user for an SRU bug if they did not provide one.
+
+    Don't ask if they tell us not to ask.
+    """
+    if not bug and not no_sru_bug:
+        bug = input(
+            "No SRU bug. Enter one now or leave blank for no SRU bug: "
+        ).strip()
+        if not bug:
+            bug = None
+    return bug
+
+
+def new_upstream_snapshot(
+    commitish: str,
+    bug: Optional[str] = None,
+    known_first_devel_upload: bool = False,
+    no_sru_bug: bool = False,
+    known_first_sru: bool = False,
+    post_patch: bool = False,
+) -> None:
+    """Perform a new upstream snapshot.
+
+    At a high level, this means:
+     - Merge the specified commitish
+     - Drop any cpick files included in the snapshot
+     - Refresh all quilt patches on the branch
+     - Update the changelog accordingly
+     - Tell user how to release this update
+    """
+    env = EnvDetails.get()
+    if not post_patch:
+        merge_commitish(commitish)
+        drop_cpicks(commitish)
+        refresh_patches(commitish)
+
+    # If arguments haven't been passed, we have a few things to determine
+    is_devel, first_devel_upload, first_sru = get_possible_devel_options(
+        known_first_devel_upload, known_first_sru, env
+    )
+    bug = get_sru_bug(bug, no_sru_bug)
+
+    update_changelog(
+        commitish, bug, env, is_devel, first_devel_upload, first_sru
+    )
+    show_release_steps(env, is_devel)
+
+
+def parse_args() -> argparse.Namespace:
+    """
+    Parsing arguments in Python
+    Is quite simple and fun
+    Just use the argparse module
+    And your script will be done!
+    """
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument(
+        "-c",
+        "--commitish",
+        required=False,
+        default="main",
+        type=str,
+        help="Commitish (tag, branch name, or commit) to merge to",
+    )
+    parser.add_argument(
+        "-b",
+        "--bug",
+        required=False,
+        type=str,
+        help="SRU bug number to add to debian/changelog",
+    )
+    parser.add_argument(
+        "-d",
+        "--first-devel-upload",
+        required=False,
+        default=False,
+        type=bool,
```

Comment:
These bool types are incorrect for argparse we should use store_true throughout otherwise the CLI prompts for a T/F value when providing those switches
```suggestion
        default=False,
        action="store_true",
```